lib/Gateway.js

```diff
@@ -1361,9 +1366,13 @@ Gateway.prototype.discoverValue = function (node, vId) {
 
         Object.assign(cfg.discovery_payload, sensor.props || {})
 
-        if (valueId.units) {
+        // Discovered valueId.unit also exists. therefore added both unit and units to be
+        // checked, but only one used
+        if (valueId.unit) {
```

Comment:
`unit` exists on the metadata, not the valueId itself. Or am I missing some zjs2mqtt magic here?

Reply:
My understanding this comes after metadata. and is parsed as valueId.

Reply:
> Or am I missing some zjs2mqtt magic here

I should create some interfaces here to make things more clear but yes the magic happens here: https://github.com/zwave-js/zwavejs2mqtt/blob/1ccab668170275f1a0195a622fe20d921bbd8de2/lib/ZwaveClient.js#L637

Reply:
What I don't understand here is why you firstly check for `valueId.unit` and then `valueId.units`? [Here](https://github.com/zwave-js/zwavejs2mqtt/blob/1ccab668170275f1a0195a622fe20d921bbd8de2/lib/ZwaveClient.js#L674) I only set `unit`

Reply:
I will fix this to just be unit! wanted to cover both cases, as it was not 100% clear for me if one case exists

Reply:
I never set `valueId.units` anywhere so that is useless :)